src/test/groovy/unit/org/pih/warehouse/requisition/RequisitionItemSpec.groovy

```diff
@@ -13,88 +16,346 @@ import spock.lang.Unroll
+        given:
+        domain.quantityApproved = quantityApproved
+        domain.productPackage = Stub(ProductPackage)
+        domain.productPackage.quantity >> productPackageQuantity
+
+        when:
+        Integer calculatedTotalQuantityApproved = domain.totalQuantityApproved()
+
+        then:
+        calculatedTotalQuantityApproved == totalQuantityApproved
+
+        where:
+        quantityApproved | productPackageQuantity || totalQuantityApproved
+        null             | null                   || 0
+        1                | null                   || 1
+        null             | 1                      || 0
+        10               | 5                      || 50
+    }
+
+    void "RequisitionItem.totalQuantityNotCanceled() should calculate #totalNotCanceledQuantity when #notCanceledQuantity and #productPackageQuantity are passed"() {
+        given:
+        RequisitionItem requisitionItem = Spy(RequisitionItem) {
+            quantityNotCanceled() >> notCanceledQuantity
+        }
+        requisitionItem.productPackage = Stub(ProductPackage)
+        requisitionItem.productPackage.quantity >> productPackageQuantity
+
+        when:
+        Integer calculatedTotalQuantityNotCanceled = requisitionItem.totalQuantityNotCanceled()
+
+        then:
+        calculatedTotalQuantityNotCanceled == totalNotCanceledQuantity
+
+        where:
+        notCanceledQuantity | productPackageQuantity || totalNotCanceledQuantity
+        null                | null                   || 0
+        1                   | null                   || 1
+        null                | 1                      || 0
+        10                  | 5                      || 50
+    }
+
+    void "RequisitionItem.isCanceled() should return: #isCanceled when #quantity and #quantityCanceled and #modificationItem and #substitutionItem and #requisitionItems are passed"() {
+        given:
+        RequisitionItem requisitionItem = Spy(RequisitionItem) {
+            totalQuantityCanceled() >> quantityCanceled
+            totalQuantity() >> quantity
+        }
+
+        requisitionItem.modificationItem >> modificationItem
+        requisitionItem.substitutionItem >> substitutionItem
+        requisitionItem.requisitionItems >> requisitionItems
+
+        when:
+        Boolean isRequisitionCanceled = requisitionItem.isCanceled()
+
+        then:
+        isCanceled == isRequisitionCanceled
+
+        where:
+        quantityCanceled | quantity | modificationItem      | substitutionItem      | requisitionItems        || isCanceled
+        3                | 3        | null                  | null                  | null                    || true
+        3                | 2        | null                  | null                  | null                    || false
+        3                | 2        | Mock(RequisitionItem) | null                  | null                    || false
```

Comment:
shouldn't quantity for these lines be 3? Otherwise they wouldn't be cancelled regardless.

Reply:
or I guess a better question is do we also want to add lines where quantity is 3 for each of these combos and assert that isCanceled is true